lib/api/readable.js

```diff
@@ -0,0 +1,256 @@
+'use strict'
```

Comment:
@nodejs/streams @jasnell @mcollina PTAL at this file and let me know if you if it might be a sensible addition to core streams?

Reply:
The idea is that we would always use this when implementing readable streams source in node and it can still be consumed in any format with close to optimal perf.

Reply:
Also it should be compatible with web body mixin.